openpype/lib/transcoding.py

```diff
@@ -1037,3 +1037,59 @@ def convert_ffprobe_fps_to_float(value):
     if divisor == 0.0:
         return 0.0
     return dividend / divisor
+
+
+def convert_colorspace_for_input_paths(
+    input_paths,
+    output_dir,
+    config_path,
+    source_color_space,
+    target_color_space,
+    logger=None
+):
+    """Convert source files from one color space to another.
+
+    Filenames of input files are kept so make sure that output directory
+    is not the same directory as input files have.
+    - This way it can handle gaps and can keep input filenames without handling
+        frame template
+
+    Args:
+        input_paths (str): Paths that should be converted. It is expected that
+            contains single file or image sequence of samy type.
+        output_dir (str): Path to directory where output will be rendered.
+            Must not be same as input's directory.
+        config_path (str): path to OCIO config file
+        source_color_space (str): ocio valid color space of source files
+        target_color_space (str): ocio valid target color space
+        logger (logging.Logger): Logger used for logging.
+
+    """
+    if logger is None:
+        logger = logging.getLogger(__name__)
+
+    input_arg = "-i"
+    oiio_cmd = [
+        get_oiio_tools_path(),
+
+        # Don't add any additional attributes
+        "--nosoftwareattrib",
+        "--colorconfig", config_path,
+        "--colorconvert", source_color_space, target_color_space
+    ]
+    for input_path in input_paths:
+        # Prepare subprocess arguments
+
+        oiio_cmd.extend([
+            input_arg, input_path,
+        ])
+
+        # Add last argument - path to output
+        base_filename = os.path.basename(input_path)
+        output_path = os.path.join(output_dir, base_filename)
+        oiio_cmd.extend([
+            "-o", output_path
+        ])
+
+        logger.debug("Conversion command: {}".format(" ".join(oiio_cmd)))
+        run_subprocess(oiio_cmd, logger=logger)
```

Comment:
This function makes more sense if it takes a single file instead of a list of filepaths.
It doesn't seem to do much optimization for running over multiple filepaths to warrant it to run over multiple files.

Additionally I think it's somewhat odd that you're specificying an output directory instead of an output filename, especially because it doesn't return the output filepaths I feel this is a bit odd.

Also other methods seem to name the color space argument as `colorspace` instead of `color_space` so I'd do that here too.

```python
convert_colorspace(filepath, out_filepath, config_path, source_colorspace, target_colorspace)
```

I'd also have it return the filepath it generated - just for clarity.

Reply:
I agree that this would be better if file path would be a string with sequential format. And not the list of strings - the docstring is stating it is str but code is expecting list. 

here is an example form [oiiotool docs](https://openimageio.readthedocs.io/en/latest/oiiotool.html#frame-sequences):
```
oiiotool big.1-3#.tif --resize 100x100 -o small.1-3#.tif
oiiotool big.1-3%04d.tif --resize 100x100 -o small.1-3%04d.tif
oiiotool --frames 1-3 big.#.tif --resize 100x100 -o small.#.tif
oiiotool --frames 1-3 big.%04d.tif --resize 100x100 -o small.%04d.tif
```

Reply:
This has not been implemented?
The method is expecting `input_paths` as list, but we need it as a single string, plus potential `--frames` argument for sequence starting at non-default start frame.